app/src/main/java/com/plenart/organizeme/fragments/MainFragment.kt

```diff
@@ -0,0 +1,151 @@
+        }
+
+    }
+
+    private fun initObservers() {
+        initUser()
+        initBoardsList()
+    }
+
+    private fun initBoardsList() {
+        viewModel.boardsList.observe(viewLifecycleOwner, Observer { boardsList ->
+            if(boardsList != null && boardsList.isNotEmpty()){
+                displayBoards(boardsList);
+            }
+            else{
+                Log.i("boardsListObserver","boardListObserver: boardsList is empty or null!")
+            }
+
+        })
+    }
+
+    private fun initUser() {
+        viewModel.user?.observe(viewLifecycleOwner, Observer { newUser ->
+            if(newUser != null){
+                updateNavigationUserDetails(newUser,true)
+            }
+            else{
+                Log.i("UserObserver","user observer: user is null")
+            }
+
+        } )
+    }
+
+    private fun updateNavigationUserDetails(loggedInUser: com.plenart.organizeme.models.User, readBoardsList: Boolean = false) {
+
+        val navView = requireActivity().findViewById<NavigationView>(R.id.nav_view)
+        val header = navView.getHeaderView(0)
+        val userImage = header.findViewById<CircleImageView>(R.id.nav_user_img)
+        val userName = header.findViewById<TextView>(R.id.tv_username)
+
+        Glide.with(this)
+            .load(loggedInUser.image)
+            .centerCrop()
+            .placeholder(R.drawable.ic_user_place_holder)
+            .into(userImage)
+
+        userName.text = loggedInUser.name
+
+        if(readBoardsList){
+            viewModel.getBoardsList()
+        }
+
+    }
+
+    private fun displayBoards(boardsList: ArrayList<Board>){
+        mainContentBinding = fragmentMainBinding.mainContentIncluded
+
+        val adapterToSet = context?.let { BoardItemsAdapter(it, boardsList) }
+        if(boardsList.size > 0){
+            mainContentBinding.apply {
+                rvBoards.visibility = View.VISIBLE
+                tvNoBoardsAvailable.visibility = View.GONE
+                tvTip.visibility = View.GONE
+                ivNoBoardsIllustration.visibility = View.GONE
+
+                rvBoards.apply {
+                    layoutManager = LinearLayoutManager(context)
+                    setHasFixedSize(true)
+                    adapter = adapterToSet
+                }
+
+            }
+
+            adapterToSet?.setOnClickListener(object: BoardItemClickInterface {
+                override fun onClick(position: Int, model: Board) {
+
+                    val directions = DrawerHostFragmentDirections.actionSecNavHostFragmentToTaskListFragment(
+                        model.documentID
+                    )
+                    requireActivity().findNavController(R.id.main_content_navigation_component).navigate(directions)
+                }
+            })
+            adapterToSet?.notifyDataSetChanged()
+        }
+        else{
+            mainContentBinding.apply {
+                rvBoards.visibility = View.GONE
```

Comment:
Just for you information: You can create a for example ViewExtension file and put there all extension functions on View.
So for visibility:
`fun View.gone() { this.visibility = View.GONE } fun View.visible() { this.visibility = View.VISIBLE }`

And then you can just call rvBoards.gone() (You can do that on all views). 
That will remove some duplicate code